src/Tribe/Aggregator/Cron.php

```diff
@@ -385,6 +386,75 @@ public function verify_child_record_creation() {
 		}
 	}
 
+	public function start_batch_pushing_records() {
+		if ( ! tribe( 'events-aggregator.main' )->is_service_active() ) {
+			return;
+		}
+
+		$records = Tribe__Events__Aggregator__Records::instance();
+
+		$query = $records->query( [
+			'post_status'    => Tribe__Events__Aggregator__Records::$status->pending,
+			'posts_per_page' => 15,
+			'order'          => 'ASC',
+			'meta_query'     => [
+				'origin-not-csv'               => [
+					'key'     => '_tribe_aggregator_origin',
+					'value'   => 'csv',
+					'compare' => '!=',
+				],
+				'batch-push-support-specified' => [
+					'key'     => '_tribe_aggregator_allow_batch_push',
+					'value'   => true,
+					'compare' => '=',
+				],
+				'batch-not-queued'             => [
+					'key'     => '_tribe_aggregator_batch_started',
+					'value'   => 'bug #23268',
+					'compare' => 'NOT EXISTS',
+				],
+			],
+			'after'          => '-4 hours',
+		] );
+
+		if ( ! $query->have_posts() ) {
+			tribe( 'logger' )->log_debug( 'No Pending Batch to be started', 'EA Cron' );
+
+			return;
+		}
+
+		tribe( 'logger' )->log_debug( "Found {$query->found_posts} records", 'EA Cron' );
+
+		$cleaner = new Tribe__Events__Aggregator__Record__Queue_Cleaner();
+		foreach ( $query->posts as $post ) {
+			$record = $records->get_by_post_id( $post );
+
+			if ( $record === null || tribe_is_error( $record ) ) {
```

Comment:
:no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: 64e9c1962bb57261fd0c186e1503b22b</sub>